internal/buildscripts/packaging/fpm/etc/otel/collector/splunk-support-bundle.ps1

```diff
@@ -0,0 +1,252 @@
+            $i = $i + 1;
+        } else {
+            usage;
+        }
+    } elseif (($args[$i] -eq "-t") -OR ($args[$i] -eq "-directory")) {
+        if (($args[$i+1]) -AND ($args[$i+1] -ne "-d") -AND ($args[$i+1] -ne "-directory") -AND ($args[$i+1] -ne "-h") -AND ($args[$i+1] -ne "-help")) {
+            $TMPDIR = $args[$i+1];
+            $i = $i + 1;
+        } else {
+            usage;
+        }
+    } elseif (($args[$i] -eq "-h") -OR ($args[$i] -eq "-help")) {
+        usage;
+    } else {
+        usage;
+    }
+}
+
+#######################################
+# Creates a unique temporary directory to store the contents of the support
+# bundle. Do not attempt to cleanup to prevent any accidental deletions.
+# This command can only be run once per second or will error out.
+# This script could result in a lot of temporary data if run multiple times.
+#  - GLOBALS: TMPDIR
+#  - ARGUMENTS: None
+#  - OUTPUTS: None
+#  - RETURN: 0 if successful, non-zero on error.
+#######################################
+function createTempDir {
+    write-host "INFO: Creating temporary directory..."
+    if (Test-Path -Path $TMPDIR) {
+        write-host "ERROR: TMPDIR ($TMPDIR) exists. Exiting."
+        exit 1
+    } else {
+        New-Item -Path $TMPDIR -ItemType Directory | Out-Null
+        New-Item -Path $TMPDIR/logs -ItemType Directory | Out-Null
+        New-Item -Path $TMPDIR/metrics -ItemType Directory | Out-Null
+        New-Item -Path $TMPDIR/zpages -ItemType Directory | Out-Null
+	# We can not create directory using special characters like : , ? 
+	# So we have encoded it and then created new directory.
+        $global:DIRECTORY = [System.Web.HTTPUtility]::UrlEncode("localhost:55679")
+        New-Item -Path $TMPDIR/zpages/$global:DIRECTORY -ItemType Directory | Out-Null
+        New-Item -Path $TMPDIR/zpages/$global:DIRECTORY/debug -ItemType Directory | Out-Null
+    }
+}
+
+#######################################
+# Gather configuration
+# Without this it is very hard to troubleshoot issues so exit if no permissions.
+#  - GLOBALS: CONFDIR, DIRECTORY, TMPDIR
+#  - ARGUMENTS: None
+#  - OUTPUTS: None
+#  - RETURN: 0 if successful, non-zero on error.
+#######################################
+function getConfig {
+    write-host "INFO: Getting configuration..."
+# If directory does not exist the support bundle is useless so exit
+    if (-NOT (Test-Path -Path $CONFDIR)) {
+        write-host "ERROR: Could not find directory ($CONFDIR)."
+        usage
+    } else {
+        Copy-Item -Path "$CONFDIR" -Destination "$TMPDIR/config" -Recurse
+    }
+# Also need to get config in memory as dynamic config may modify stored config
+# It's possible user has disabled collecting in memory config
+    $connection = New-Object System.Net.Sockets.TcpClient("localhost", 55554)
+    if ($connection.Connected) {
+        cmd.exe /c "curl -s http://localhost:55554/debug/configz/initial > $TMPDIR/config/initial.yaml 2>&1"

```

Comment:
Is curl guaranteed to exist -- this script needs to support all versions of Windows we support? Perhaps native powershell commands should be used here? https://stackoverflow.com/questions/2710748/run-curl-commands-from-windows-console

Reply:
We have changed the curl command with the Invoke-WebRequest command that is supported for all the versions of windows server 2012 and later.
![image](https://user-images.githubusercontent.com/89311282/130462782-42910f06-5261-4b92-9fff-b2c96e1ad6a5.png)